### PR TITLE
Fix invalid value for weights_summary

### PR DESCRIPTION
**Pull request by rohitgr7**
## What does this PR do?

The main fix here is if you do `model.summarize(None)` it still prints the total params, trainable params, ... and if you do `model.summarize('invalid_string or value')` it won't raise an error and will act like `None`.

## Before submitting
- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes [if needed]?
- [ ] Did you write any new necessary tests [no need for typos, docs]? 
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:  

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified
 - [x] **Check that target branch and milestone are aligned!**
 

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by Borda** [comment]
@rohitgr7 as you are changing API we need to move it to 1.2
mind rebase on feature 1.2 🐰

**Comment by rohitgr7** [comment]
I thought I created this branch on top of release branch only. Will update it.

**Review by carmocca — APPROVED** [review]
LGTM. One thing:

Why not keep `mode` to avoid the API change?

**Comment by awaelchli** [comment]
+1 for keeping mode please :)

**Comment by rohitgr7** [comment]
I did that to have the exception at one place for both. Will update this then.

**Comment by Borda** [comment]
@rohitgr7 is it as a bugfix for master or a feature for 1.2? there is some confusion as it is a 1.2 milestone but the target is `master`

**Comment by rohitgr7** [comment]
bugfix now. forgot to update the milestone. Updated!

**Comment by carmocca** [comment]
> I did that to have the exception at one place for both the cases.

I like what you proposed. We can do it in another PR

**Review by carmocca — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]

File: pytorch_lightning/core/lightning.py [hunk 1]
```diff
@@ -1328,8 +1328,16 @@ def tbptt_split_batch(self, batch, split_size):
         return splits
 
     def summarize(self, mode: str = ModelSummary.MODE_DEFAULT) -> ModelSummary:
```
[the remark this answers is not in the document]
Comment (carmocca) [reply pytorch_lightning/core/lightning.py:1330]: The logic will be there, just outside the function wherever it is called. IMO it's convenient to allow it.

[the remark this answers is not in the document]
Comment (awaelchli) [reply pytorch_lightning/core/lightning.py:1330]: then you need to put the type annotation for the optional arg and optional return

[the remark this answers is not in the document]
Comment (rohitgr7) [reply pytorch_lightning/core/lightning.py]: done

**Review by tchaton — APPROVED** [review]

**Pull request merged**